### ISSUE TITLE
reference:file: bundles can not be loaded if org.knopflerfish.osgi.registerserviceurlhandler=false

### ISSUE THREAD

**Issue by dpolivaev** [knopflerfish/knopflerfish.org#47]
freeplane uses knopflerfish as OSGi framework.
Recently it was added an API which allows to embed it into java applications.
One special use case is running in servlet to perform calculations defined in mind maps.

Because URL.setURLStreamHandlerFactory is already used by Tomcat ( embedded in spring boot)
I used configuration option `org.knopflerfish.osgi.registerserviceurlhandler=false`
Unfortunately as a consequence protocol 'reference' was no more resolved.

However it is needed at `Bundles.install0` which tries to create the bundle URL calling ` URL url = new URL(location);`

~~~~
 BundleImpl install0(String location, InputStream in, Object checkContext, Bundle caller)
    throws BundleException {
    InputStream bin;
    BundleArchive ba = null;
    try {
      if (in == null) {
        // Do it the manual way to have a chance to
        // set request properties
        final URL url  = new URL(location);
        final URLConnection conn = url.openConnection();
~~~~

As a work around I added following static block to one of the components:
~~~~
       try {
            Constructor<ServiceURLStreamHandlerFactory> constructor = ServiceURLStreamHandlerFactory.class.getDeclaredConstructor();
            constructor.setAccessible(true);
            ServiceURLStreamHandlerFactory systemUrlStreamHandlerFactory = constructor.newInstance();
            TomcatURLStreamHandlerFactory.getInstance().addUserFactory(systemUrlStreamHandlerFactory);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
~~~~

It creates ServiceURLStreamHandlerFactory  from knopflerfish and registers it as user factory by TomcatURLStreamHandlerFactory. 

If the constructor was public or I could at least access `FrameworkContext.systemUrlStreamHandlerFactory` or some static method to create it I would not need the trick with the reflection.

It would be even better if the correct handler was used directly at the position where URL is created.

Could you consider any of that?
Let me know if I can help by submitting a pull request.

**Comment by janste63** [comment]
I see your need and it seems reasonable. I saw your pull request and will check it later this week or early next week.

      Thanks,
         Jan

**Comment by janste63** [comment]
Seems ok, I've merged it (#48). Thanks for the improvement.
